lxc-to-lxd/main_migrate.go

```diff
@@ -0,0 +1,602 @@
+	req.Profiles = []string{"default"}
+
+	// Set the container architecture if set in LXC
+	fmt.Println("Processing container architecture configuration")
+	var arch string
+	value = getConfig(conf, "lxc.arch")
+	if value == nil {
+		fmt.Println("Couldn't find container architecture, assuming native")
+		arch = runtime.GOARCH
+	} else {
+		arch = value[0]
+	}
+
+	archId, err := osarch.ArchitectureId(arch)
+	if err != nil {
+		return err
+	}
+
+	req.Architecture, err = osarch.ArchitectureName(archId)
+	if err != nil {
+		return err
+	}
+
+	if storage != "" {
+		req.Devices["root"] = map[string]string{
+			"type": "disk",
+			"pool": storage,
+			"path": "/",
+		}
+	}
+
+	if debug {
+		out, _ := json.MarshalIndent(req, "", "  ")
+		fmt.Printf("LXD container config:\n%v\n", string(out))
+	}
+
+	// Create container
+	fmt.Println("Creating container")
+	if dryRun {
+		fmt.Println("Would create container now")
+	} else {
+		op, err := d.CreateContainer(req)
+		if err != nil {
+			return err
+		}
+
+		progress := utils.ProgressRenderer{Format: "Transferring container: %s"}
+		_, err = op.AddHandler(progress.UpdateOp)
+		if err != nil {
+			progress.Done("")
+			return err
+		}
+
+		rootfs, _ := getRootfs(conf)
+
+		err = transferRootfs(d, op, rootfs, rsyncArgs)
+		if err != nil {
+			return err
+		}
+
+		progress.Done(fmt.Sprintf("Container '%s' successfully created", container.Name()))
+	}
+
+	return nil
+}
+
+func convertNetworkConfig(container *lxc.Container, devices types.Devices) error {
+	networkDevice := func(network map[string]string) map[string]string {
+		if network == nil {
+			return nil
+		}
+
+		device := make(map[string]string, 0)
+		device["type"] = "nic"
+
+		// Get the device type
+		device["nictype"] = network["type"]
+
+		// Convert the configuration
+		for k, v := range network {
+			switch k {
+			case "hwaddr", "mtu", "name":
+				device[k] = v
+			case "link":
+				device["parent"] = v
+			case "veth_pair":
+				device["host_name"] = v
+			case "":
+				// empty key
+				return nil
+			}
+		}
+
+		switch device["nictype"] {
+		case "veth":
+			_, ok := device["parent"]
+			if ok {
+				device["nictype"] = "bridged"
+			} else {
+				device["nictype"] = "p2p"
+			}
+		case "phys":
+			device["nictype"] = "physical"
+		case "empty":
+			return nil
+		}
+
+		return device
+	}
+
+	fmt.Println("Processing network configuration")
+
+	devices["eth0"] = make(map[string]string, 0)
+	devices["eth0"]["type"] = "none"
+
+	// New config key
+	for i := range container.ConfigItem("lxc.net") {
+		network := networkGet(container, i, "lxc.net")
+
+		dev := networkDevice(network)
+		if dev == nil {
+			continue
+		}
+
+		devices[fmt.Sprintf("convert_net%d", i)] = dev
+	}
+
+	// Old config key
+	for i := range container.ConfigItem("lxc.network") {
+		network := networkGet(container, i, "lxc.network")
+
+		dev := networkDevice(network)
+		if dev == nil {
+			continue
+		}
+
+		devices[fmt.Sprintf("convert_net%d", len(devices))] = dev
```

Comment:
`convert_net`? Why though? :)

Reply:
This was taken directly from the `lxc-to-lxd` script.

Reply:
Ok, I'd still prefer this to simply be `net%d` I think. :)